lib/trip_dispatcher.rb

```diff
@@ -27,14 +33,68 @@ def inspect
               #{passengers.count} passengers>"
     end
 
+    def available_drivers
+      available_drivers_list = []
+      @drivers.each do |driver|
+        if driver.status == :AVAILABLE && (driver.trips.each { |trip| trip.end_time != nil })
+          available_drivers_list << driver
+        end
+      end 
+      return available_drivers_list
+    end 
+
+    def pick_most_eligible_driver(available_drivers_list)
+      longest_ago = Time.now
+      best_driver = nil
+      available_drivers_list.each do |driver|
+      # Pick first driver in array of available drivers that has nil or no trips
+        if driver.trips == nil || driver.trips == []
```

Comment:
Can be made a little clearer with:

```suggestion
        if driver.trips.nil? || driver.trips.empty?
```